drivers/i2c/i2c_nrfx_twi.c

```diff
@@ -136,29 +136,29 @@ static int init_twi(struct device *dev, const nrfx_twi_config_t *config)
 	return 0;
 }
 
-#define I2C_NRFX_TWI_DEVICE(idx)					\
-	static int twi_##idx##_init(struct device *dev)			\
-	{								\
-		IRQ_CONNECT(DT_I2C_##idx##_IRQ,			\
-			    CONFIG_I2C_##idx##_IRQ_PRI,			\
-			    nrfx_isr, nrfx_twi_##idx##_irq_handler, 0);	\
-		const nrfx_twi_config_t config = {			\
-			.scl       = DT_I2C_##idx##_SCL_PIN,	\
-			.sda       = DT_I2C_##idx##_SDA_PIN,	\
-			.frequency = NRF_TWI_FREQ_100K,			\
-		};							\
-		return init_twi(dev, &config);				\
-	}								\
-	static struct i2c_nrfx_twi_data twi_##idx##_data = {		\
-		.sync = _K_SEM_INITIALIZER(twi_##idx##_data.sync, 0, 1)	\
-	};								\
-	static const struct i2c_nrfx_twi_config twi_##idx##_config = {	\
-		.twi = NRFX_TWI_INSTANCE(idx)				\
-	};								\
-	DEVICE_AND_API_INIT(twi_##idx, CONFIG_I2C_##idx##_NAME,		\
-			    twi_##idx##_init, &twi_##idx##_data,	\
-			    &twi_##idx##_config, POST_KERNEL,		\
-			    CONFIG_I2C_INIT_PRIORITY,			\
+#define I2C_NRFX_TWI_DEVICE(idx)					   \
+	static int twi_##idx##_init(struct device *dev)			   \
+	{								   \
+		IRQ_CONNECT(DT_NORDIC_NRF_I2C_I2C_##idx##_IRQ,		   \
+			    DT_NORDIC_NRF_I2C_I2C_##idx##_IRQ_PRIORITY,	   \
+			    nrfx_isr, nrfx_twi_##idx##_irq_handler, 0);	   \
+		const nrfx_twi_config_t config = {			   \
+			.scl       = DT_NORDIC_NRF_I2C_I2C_##idx##_SCL_PIN,\
+			.sda       = DT_NORDIC_NRF_I2C_I2C_##idx##_SDA_PIN,\
+			.frequency = NRF_TWI_FREQ_100K,			   \
+		};							   \
+		return init_twi(dev, &config);				   \
+	}								   \
+	static struct i2c_nrfx_twi_data twi_##idx##_data = {		   \
+		.sync = _K_SEM_INITIALIZER(twi_##idx##_data.sync, 0, 1)	   \
+	};								   \
+	static const struct i2c_nrfx_twi_config twi_##idx##_config = {	   \
+		.twi = NRFX_TWI_INSTANCE(idx)				   \
+	};								   \
+	DEVICE_AND_API_INIT(twi_##idx, DT_NORDIC_NRF_I2C_I2C_##idx##_LABEL,\
```

Comment:
technically I wouldn't change the name right now since sensor drivers still use CONFIG_I2C_x_NAME.  But I'm ok if you want to.